Frends.AzureBlobStorage.ReadBlob/Frends.AzureBlobStorage.ReadBlob/Definitions/Source.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+#pragma warning disable 1591
+namespace Frends.AzureBlobStorage.ReadBlob
+{
+    public class Source
+    {
+        /// <summary>
+        ///     Use either Uri+SAS Token or Connection string
+        /// </summary>
+        [DefaultValue("https://xx.blob.xx.xx.net/")]
+        [DisplayFormat(DataFormatString = "Text")]
+        public string Uri { get; set; }
+
+        [DisplayFormat(DataFormatString = "Text")]
+        public string SasToken { get; set; }
+
+        /// <summary>
+        ///     Use either Uri+SAS Token or Connection string
+        /// </summary>
+        [DisplayFormat(DataFormatString = "Text")]
+        public string ConnectionString { get; set; }
+
+        /// <summary>
+        ///     Name of the azure blob storage container where the file is downloaded from.
+        /// </summary>
+        [DisplayFormat(DataFormatString = "Text")]
+        public string ContainerName { get; set; }
+
+        /// <summary>
+        ///     Name of the blob to download
+        /// </summary>
+        [DisplayFormat(DataFormatString = "Text")]
+        public string BlobName { get; set; }
+
+        [DefaultValue("UTF8")]
+        [DisplayFormat(DataFormatString = "Text")]
+        public Encode Encoding { get; set; }
+    }
+}
```

Comment:
Since there are multiple different authentication methods, you should have a selector where user is able to select which authentication he wants to use and then display only fields related to that authentication method. Example: https://github.com/FrendsPlatform/Frends.Salesforce/blob/main/Frends.Salesforce.ExecuteQuery/Frends.Salesforce.ExecuteQuery/Definitions/Options.cs#L9. You should also use the authentication options to different Options-class.

Reply:
You should still move the authentication related stuff to separate Options-class.